src/condor_utils/proc_family_direct_cgroup_v1.cpp

```diff
@@ -0,0 +1,485 @@
+		return false;
+	}
+
+	uint64_t user_usec = 0;
+	uint64_t sys_usec  = 0;
+
+	char word[128]; // max size of a word in cpu.stat
+	while (fscanf(f, "%s", word) != EOF) {
+		// if word is usage_usec
+		if (strcmp(word, "user") == 0) {
+			// next word is the user time in microseconds
+			if (fscanf(f, "%ld", &user_usec) != 1) {
+				dprintf(D_ALWAYS, "Error reading user_usec field out of cpu.stat\n");
+				fclose(f);
+				return false;
+			}
+		}
+		if (strcmp(word, "system") == 0) {
+			// next word is the system time in microseconds
+			if (fscanf(f, "%ld", &sys_usec) != 1) {
+				dprintf(D_ALWAYS, "Error reading system_usec field out of cpu.stat\n");
+				fclose(f);
+				return false;
+			}
+		}
+	}
+	fclose(f);
+
+	time_t wall_time = time(nullptr) - start_time;
+	usage.percent_cpu = double(user_usec + sys_usec) / double((wall_time * 1'000'000));
+
+	usage.user_cpu_time = user_usec / 1'000'000; // usage.user_cpu_times in seconds, ugh
+	usage.sys_cpu_time  =  sys_usec / 1'000'000; //  usage.sys_cpu_times in seconds, ugh
+
+	stdfs::path memory_current = cgroup_root_dir / "memory" / cgroup_name / "memory.usage_in_bytes";
+	stdfs::path memory_peak   = cgroup_root_dir / "memory" / cgroup_name / "memory.max_usage_in_bytes";
+
+	f = fopen(memory_current.c_str(), "r");
+	if (!f) {
+		dprintf(D_ALWAYS, "ProcFamilyDirectCgroupV1::get_usage cannot open %s: %d %s\n", memory_current.c_str(), errno, strerror(errno));
+		return false;
+	}
+
+	uint64_t memory_current_value = 0;
+	if (fscanf(f, "%ld", &memory_current_value) != 1) {
+		dprintf(D_ALWAYS, "ProcFamilyDirectCgroupV1::get_usage cannot read %s: %d %s\n", memory_current.c_str(), errno, strerror(errno));
+		fclose(f);
+		return false;
+	}
+	fclose(f);
+
+	uint64_t memory_peak_value = 0;
+
+	f = fopen(memory_peak.c_str(), "r");
+	if (!f) {
+		// Some cgroup v1 versions don't have this file
+		dprintf(D_ALWAYS, "ProcFamilyDirectCgroupV1::get_usage cannot open %s: %d %s\n", memory_peak.c_str(), errno, strerror(errno));
+	} else {
+		if (fscanf(f, "%ld", &memory_peak_value) != 1) {
+
+			// But this error should never happen
+			dprintf(D_ALWAYS, "ProcFamilyDirectCgroupV1::get_usage cannot read %s: %d %s\n", memory_peak.c_str(), errno, strerror(errno));
+			fclose(f);
+			return false;
+		}
+		fclose(f);
+	}
+
+	// usage is in kbytes.  cgroups reports in bytes
+	usage.total_image_size = usage.total_resident_set_size = (memory_current_value / 1024);
+
+	// Sanity check if system is missing memory.peak file
+	if (memory_current_value > memory_peak_value) {
+		memory_peak_value = memory_current_value;
+	}
+
+	// More sanity checking to latch memory high water mark
+	if ((memory_peak_value / 1024) > usage.max_image_size) {
+		usage.max_image_size = memory_peak_value / 1024;
+	}
+
+	return true;
+}
+
+// Note that in cgroup v1, cgroup.procs contains only those processes in
+// this direct cgroup, and does not contain processes in any descendent
+// cgroup (except the / croup, which does).
+	bool
```

Comment:
```suggestion
bool
```

... as long as I'm suggesting thirty billion changes already...